File: test/corpus/collections/map_closures.exs
defmodule RustQ.Corpus.Collections.MapClosures do
  use RustQ.Meta

  alias RustQ.Type, as: R

  @spec increment(R.u32()) :: R.u32()
  defrust(increment(value), do: value + 1)

  @spec map_named(R.vec(R.u32())) :: R.vec(R.u32())
  defrust map_named(values) do
    Enum.map(values, &increment/1)
  end

  @spec map_block(R.vec(R.u32())) :: R.vec(R.u32())
  defrust map_block(values) do
    Enum.map(values, fn value ->
      doubled = value * 2
      doubled + 1
    end)
  end
end

File: test/corpus/collections/map_closures.rs
fn increment(value: u32) -> u32 {
    value + 1
}

fn map_named(values: Vec<u32>) -> Vec<u32> {
    values.into_iter().map(increment).collect::<Vec<u32>>()
}

fn map_block(values: Vec<u32>) -> Vec<u32> {
    values
        .into_iter()
        .map(|value| {
            let doubled = value * 2;
            doubled + 1
        })
        .collect::<Vec<u32>>()
}
